include/ignition/physics/detail/FeatureList.hh

```diff
@@ -367,9 +502,50 @@ namespace ignition
       template <template<typename> class Selector, typename FeatureListT>
       struct ExtractAPI
       {
-        public: template<typename... T>
-        class type : public virtual
-            Aggregate<Selector, FeatureListT>::template type<T...> { };
+        template <typename FeatureTuple, typename... T>
```

Comment:
Can the `Aggregate` template be removed now that `ExtractAPI` doesn't use it?

Reply:
Thanks for the review, @azeey ! I'll go ahead and merge this PR to unblock `ign-gazebo`, but @mxgrey , feel free to follow up in a subsequent PR if needed.

Reply:
> From what I can tell, this still checks for duplicate Features, but I remember you mentioned that in order to reduce compile times, we have to assume that there are no duplicate Features in a FeatureList. Is that still the case?

That is not the case, fortunately! There should be no problem at all with having duplicate features.

> Can ComposePlugin be updated to work like the new ExtractAPI? Would that save any more compile time?

I considered that, but I didn't have enough time to really explore it. When I inspect the symbols that get generated, there are relatively few symbols related to `ComposePlugin`, most likely because all entity types share the same plugin type, so cutting down on `ComposePlugin` probably won't offer much. It's certainly worth trying if someone has cycles to spare.

> Can the Aggregate template be removed now that ExtractAPI doesn't use it?

For some reason I thought `Aggregate` was still being used somewhere, but I can no longer find wherever I thought it was being used. So sure, I guess we can throw it out now.